src/samVisualize/commands/visualizeSamTemplate/samVisualization.ts

```diff
@@ -0,0 +1,227 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { basename, join } from 'path'
+import { generateGraphFromYaml } from '../../graphGeneration/cfnTemplateGraphGenerator'
+import { generateResourceLineMap, ResourceLineMap } from '../../rendering/navigation'
+import { readFileSync } from 'fs-extra'
+import * as vscode from 'vscode'
+import * as _ from 'lodash'
+import { generateIconsMap } from '../../rendering/icons'
+export class SamVisualization {
+    /**
+     * The webviewPanel holding the visualization
+     */
+    public webviewPanel: vscode.WebviewPanel | undefined
+
+    /**
+     * The URI associated with the TextDocument containing the template being rendered
+     */
+    public readonly textDocumentUri: vscode.Uri
+
+    private textDocument: vscode.TextDocument
+
+    private resourceLineMap: ResourceLineMap | undefined
+
+    public disposables: vscode.Disposable[] = []
+
+    // Used to establish paths to resources that the webview uses
+    private readonly extensionContext: vscode.ExtensionContext
+
+    // To let the extension know a visualization has been closed
+    private readonly onVisualizationDisposeEmitter = new vscode.EventEmitter<void>()
+
+    constructor(textDocument: vscode.TextDocument, extensionContext: vscode.ExtensionContext) {
+        this.extensionContext = extensionContext
+        this.textDocument = textDocument
```

Comment:
Nits:
* Declare constructor as `public`
* Unmodified class variable assignments can be done in the constructor:
```
public constructor (
    private textDocument: vscode.TextDocument
    private extensionContext: vscode.ExtensionContext
)
```